log4j-core/src/main/java/org/apache/logging/log4j/core/appender/rolling/DefaultRolloverStrategy.java

```diff
@@ -83,6 +85,195 @@
     private static final int DEFAULT_WINDOW_SIZE = 7;
 
     /**
+     * Builds DefaultRolloverStrategy instances.
+     */
+    public static class Builder implements org.apache.logging.log4j.core.util.Builder<DefaultRolloverStrategy> {
+        @PluginBuilderAttribute("max")
+        private String max;
+        
+        @PluginBuilderAttribute("min")
+        private String min;
+        
+        @PluginBuilderAttribute("fileIndex")
+        private String fileIndex;
+
+        @PluginBuilderAttribute("compressionLevel")
+        private String compressionLevelStr;
+
+        @PluginElement("Actions")
+        private Action[] customActions;
+
+        @PluginBuilderAttribute(value = "stopCustomActionsOnError")
+        private boolean stopCustomActionsOnError = true;
+
+        @PluginBuilderAttribute(value = "compressTmpFilePattern")
```

Comment:
I like the original name from the ticket better: "temporaryCompressedFilePattern". I do not think we use "tmp" much if at all. IMO, if we want short "temp" would be friendlier: "tempCompressedFilePattern",